### ISSUE TITLE
Could not derive type of path parameter "id"

### ISSUE THREAD

**Issue by KOliver94** [tfranzel/drf-spectacular#999]
Hey everyone,

First of all, thanks for this amazing package! I would like to ask for some help.

**Describe the bug**
I have nested paths but **I don't use** `drf-nested-routers` because I thought my needs can be satisfied without it. However, I'm getting `could not derive type of path parameter "id" because it is untyped and obtaining queryset from the viewset failed. Consider adding a type to the path (e.g. <int:id>) or annotating the parameter type with @extend_schema. Defaulting to "string".` errors on those paths.

I noticed that if I have the `.filter(request=self.kwargs["request_pk"])` or similar call it thorws this warning but if I'm using `Comment.objects.all()` everything is fine _(except it's not doing what I want)_.

**To Reproduce**
urls.py
```
router = SimpleRouter(trailing_slash=False)
router.register(r"requests", RequestAdminViewSet, basename="request")

request_router = SimpleRouter(trailing_slash=False)
request_router.register(r"comments", CommentAdminViewSet, basename="comment")

urlpatterns = [
    path("", include(router.urls)),
    path(
        "requests/<int:request_pk>/",
        include((request_router.urls, "request"), namespace="request"),
    ),
]
```
views.py
```

class CommentAdminViewSet(ModelViewSet):
    ...

    def get_queryset(self):
        return Comment.objects.filter(
            request=get_object_or_404(Request, pk=self.kwargs["request_pk"])
        )
```

**Expected behavior**
It should not throw warnings.

What do you think what do I do wrong and what's the easiest way to fix? Thanks!

**Comment by tfranzel** [comment]
Hi, thanks! 

this is covered by several tests and should work like that:`"requests/<int:request_pk>/"`. It will get picked up and discovery will stop there before raising an exception.

However, your warning is regarding a parameter `id` not `request_pk`! On more recent versions there should also be a trace to the offending view/serializer preceding the warning.

> I noticed that if I have the .filter(request=self.kwargs["request_pk"]) or similar call it thorws this warning but if I'm using Comment.objects.all() everything is fine (except it's not doing what I want).

Since it was not able to find anything, spectacular will attempt to look in the queryset/model also, but even if this were not to fail, I suppose that parameter can't be found there.

Still, if you want to prevent that exception, you don't need to do `all()`. Here is a special construction with `none()` for schema generation runs only: https://drf-spectacular.readthedocs.io/en/latest/faq.html#my-get-queryset-depends-on-some-attributes-not-available-at-schema-generation-time that should fix that part at least.

**Comment by tfranzel** [comment]
I worded this a bit weird.

tldr: keep the `<int:request_pk>/`, it already works. Also follow the FAQ entry. It will allow to successfully retrieve the type of `id` from `Comment` (through an empty queryset) without failing.

**Comment by KOliver94** [comment]
Oh I see. Thank you for the help and the explanation!
